rider-fsharp/src/main/java/com/jetbrains/rider/plugins/fsharp/completion/FSharpCompletionUtils.kt

```diff
@@ -0,0 +1,27 @@
+package com.jetbrains.rider.plugins.fsharp.completion
+
+import com.intellij.psi.PsiWhiteSpace
+import com.jetbrains.rider.ideaInterop.fileTypes.fsharp.lexer.FSharpTokenType
+import com.jetbrains.rider.ideaInterop.fileTypes.fsharp.psi.FSharpFile
+import com.jetbrains.rider.ideaInterop.fileTypes.fsharp.psi.FSharpStringLiteralExpression
+import com.intellij.psi.util.elementType
+import com.jetbrains.rider.ideaInterop.fileTypes.fsharp.psi.FSharpStringLiteralType
+
+// check for #r "... <caret> ..." 
+fun insideReferenceDirective(file: FSharpFile, offset: Int): Boolean {
+  val string = file.findElementAt(offset)?.parent
+  if (string !is FSharpStringLiteralExpression) return false
+
+  return when (string.literalType) {
+    FSharpStringLiteralType.RegularString,
+    FSharpStringLiteralType.VerbatimString,
+    FSharpStringLiteralType.TripleQuoteString -> {
+      val whiteSpace = string.prevSibling
+      if (whiteSpace !is PsiWhiteSpace) return false
```

Comment:
Is there an API to get previous meaningful token/sibling node, skipping whitespaces, comments and other ignored nodes?

Here're some edge case examples using it would probably fix:

```fsharp
#r""
```

```fsharp
#r
    ""
```